File: src/demo02_talker_listener/src/person_listener.cpp

#include "ros/ros.h"
#include "demo02_talker_listener/Person.h"

void do_msg(const demo02_talker_listener::Person::ConstPtr& peo)
{
    ROS_INFO("%s %d %f", peo->name.c_str(), peo->age, peo->height);
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "person_listener");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe<demo02_talker_listener::Person>("person_chatter",10,do_msg);
    ros::spin();
    return 0;
}

File: src/demo02_talker_listener/scripts/person_listener.py
#! /usr/bin/env python
import rospy
from demo02_talker_listener.msg import Person

def do_msg(peo):
    rospy.loginfo(f'{peo.name} {peo.age} {peo.height}')

if __name__ == "__main__":
    rospy.init_node("person_listener")
    sub = rospy.Subscriber("person_chatter", Person, do_msg)
    rospy.spin()
